packages/kotti-ui/source/kotti-field-file-upload/validators.ts

```diff
@@ -0,0 +1,50 @@
+import { KottiFieldFileUpload } from './types'
+
+export const isSelectingMultipleFilesWhenNotAllowed = (
+	allowMultiple: boolean,
+	selectionSize: number,
+): boolean => !allowMultiple && selectionSize > 1
+
+const isValidFileSize = (fileSize: number, maxFileSize: number): boolean =>
+	fileSize < maxFileSize
+
+const isValidFileExtension = (
+	fileName: string,
+	validFileExtensions: Array<string>,
+): boolean => {
+	if (validFileExtensions.length === 0) return true
+
+	const fileExtension = (fileName.split('.').pop() ?? '').trim().toUpperCase()
+
+	if (!fileExtension) return false
+
+	return validFileExtensions
+		.map((extension) => extension.trim().toUpperCase())
+		.includes(fileExtension)
+}
+
+export const isValidUploadedFileStatus = (
+	valueItem: KottiFieldFileUpload.ValueInternal[0],
+) =>
+	valueItem.status === KottiFieldFileUpload.Status.Common.INVALID ||
```

Comment:
I thought it's called UPLOADED_WITH_ERROR

@santiagoballadares

Reply:
I saw the usage of this later

looks to me like you intentionally want to return true even if the extension and size validators fail

which doesn't comply with the last discussion we had about UPLOADED_WITH_ERROR for already uploaded files